addon/lint/lint.js

```diff
@@ -187,7 +187,7 @@
   CodeMirror.defineOption("lint", false, function(cm, val, old) {
     if (old && old != CodeMirror.Init) {
       clearMarks(cm);
-      if (!state.options.lintOnChange !== false)
+      if (cm.state.lint.options.lintOnChange === undefined || cm.state.lint.options.lintOnChange === true)
```

Comment:
I was very confused as a programmer so I wrote this program:
```javascript
a = {};
// lintOnChange undefined
console.log(!a.lintOnChange !== false);
a.lintOnChange = true;
console.log(!a.lintOnChange !== false);
a.lintOnChange = false;
console.log(!a.lintOnChange !== false);
```
which generates this:
```
true
false
true
```

Am I missing anything?